examples/gno.land/p/demo/avl/pager/pager.gno

```diff
@@ -0,0 +1,239 @@
+package pager
+
+import (
+	"math"
+	"net/url"
+	"strconv"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+// Pager is a struct that holds the AVL tree and pagination parameters.
+type Pager struct {
+	Tree            *avl.Tree
+	PageQueryParam  string
+	SizeQueryParam  string
+	DefaultPageSize int
+}
+
+// Page represents a single page of results.
+type Page struct {
+	Items      []Item
+	PageNumber int
+	PageSize   int
+	TotalItems int
+	TotalPages int
+	HasPrev    bool
+	HasNext    bool
+	Pager      *Pager // Reference to the parent Pager
+}
+
+// Item represents a key-value pair in the AVL tree.
+type Item struct {
+	Key   string
+	Value interface{}
+}
+
+// NewPager creates a new Pager with default values.
+func NewPager(tree *avl.Tree, defaultPageSize int) *Pager {
+	return &Pager{
+		Tree:            tree,
+		PageQueryParam:  "page",
+		SizeQueryParam:  "size",
+		DefaultPageSize: defaultPageSize,
+	}
+}
+
+// GetPage retrieves a page of results from the AVL tree.
+func (p *Pager) GetPage(pageNumber int) *Page {
+	return p.GetPageWithSize(pageNumber, p.DefaultPageSize)
+}
+
+func (p *Pager) GetPageWithSize(pageNumber, pageSize int) *Page {
+	totalItems := p.Tree.Size()
+	totalPages := int(math.Ceil(float64(totalItems) / float64(pageSize)))
+
+	if pageSize < 1 {
+		return &Page{
+			Items:      []Item{},
+			PageNumber: 0,
+			PageSize:   0,
+			TotalItems: totalItems,
+			TotalPages: totalPages,
+			HasPrev:    false,
+			HasNext:    false,
+			Pager:      p,
+		}
+	}
+
+	if pageNumber < 1 {
+		return &Page{
+			Items:      []Item{},
+			PageNumber: 0,
+			PageSize:   pageSize,
+			TotalItems: totalItems,
+			TotalPages: totalPages,
+			HasPrev:    false,
+			HasNext:    pageNumber < totalPages,
+			Pager:      p,
+		}
+	}
+
+	if pageNumber > totalPages {
+		return &Page{
+			Items:      []Item{},
+			PageNumber: pageNumber,
+			PageSize:   pageSize,
+			TotalItems: totalItems,
+			TotalPages: totalPages,
+			HasPrev:    totalPages > 0,
+			HasNext:    false,
+			Pager:      p,
+		}
+	}
+
+	startIndex := (pageNumber - 1) * pageSize
+	endIndex := startIndex + pageSize
+	if endIndex > totalItems {
+		endIndex = totalItems
+	}
+
+	items := []Item{}
+	p.Tree.ReverseIterateByOffset(startIndex, endIndex-startIndex, func(key string, value interface{}) bool {
+		items = append(items, Item{Key: key, Value: value})
+		return false
```

Comment:
offtopic: a bit misleading, normally, if you want to continue, you return true, but it is what it is.